File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        int[] size = new int[nums.length];
        int max = 0;
        for (int i = 0; i < nums.length; i++) {
            size[i] = 1; 
            for (int j = i; j >= 0; j--) {
                if (nums[j] < nums[i]) {
                    size[i] = Math.max(size[i], size[j] + 1);
                }
            }
            max = Math.max(max, size[i]);
        }
        return max;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        product = 1

        zero_count = 0

        for num in nums:
            if num != 0:
                product *= num
            if num == 0:
                zero_count += 1
        
        res = [0] * len(nums)

        if zero_count > 1:
            return res

        for idx, num in enumerate(nums):
            if num == 0:
                res[idx] = product
            elif zero_count == 1:
                res[idx] = 0
            else:
                res[idx] = math.floor(product / num)
        
        return res